### ISSUE TITLE
Live Video Playlist

### ISSUE THREAD

**Issue by mikeyounge** [meeb/tubesync#563]
I am not sure how common this is, but I have a channel that is doing a bunch of live streams this week and next week. They created a playlist of their upcoming schedule. The streams started airing today in 3 hour blocks. I set it up as a playlist channel in Tubesync and it populated with 43 upcoming videos. The first video was live when I created the channel. It saw all of the metadata and started to try to pull thumbnails. At first it wasn't successful, but on the auto-retry it worked. After that video was over, the next 3 hour block started. I set the update time to be every three hours and it ran fine, but only pulled the thumbnail for the new live video. It did not go back and start the download of the past video. It came back with the status " Media cannot be downloaded because it has no formats which match the source requirements." I have the format set to pull any video, regardless of quality. I manually skipped the media and re-queued it and it downloaded fine in HD. After the 2 block finished and the channel was refreshed, the same thing happened: downloaded the thumbnail and metadata fine for the live video, but did not go back to grab the previous block. I was again able to manually skip and re-queue with success. I have noticed this behavior on other live videos, but thought it might be the creators because they aren't tech-savvy people. Today's experiment makes me think it more about how Tubesync treats live streams. Any thoughts on this use-case and what is going on? Let me know the easiest method for pulling logs that will be useful to you and I'm happy to provide. Thanks!

Channel Playlist Key: PL3XZNMGhpynMZGjl0bdn5M_s9EVSsOhAH

**Comment by meeb** [comment]
If I understand your issue correctly your playlist contains items that aren't available yet and will be available at some point in the future. They are then retried over a day or so, and then marked as failed because they couldn't be downloaded? The error (no formats available) makes sense because when tubesync attempted to download them obviously they didn't have any formats available as it wasn't uploaded as a VOD yet.

Yes, this would be "fixed" by skipping then unskipping the items as this resets the download attempts.

There currently isn't any obvious way to tell if a media item will be available in the future of it it's just permanently failed so short of retrying failed downloads a lot more (up to a week or so) I can't see a sensible way to quickly fix this.

In the short term, you'll need to skip then unskip failed media items to work around this, sorry. No-one else has mentioned this issue to date so while I'm sympathetic and will look into a solution it doesn't seem very common.

**Comment by mikeyounge** [comment]
Maybe on the next scan a check can be put on files with the specific error of "no formats" and if the error could be corrected then it can retry it. I'm not sure how those are logged, so I don't know how simple it would be.

**Comment by meeb** [comment]
The "no formats" error can occur for a number of permanent failure reasons as well. As far as I can see right now there's no really any way to solve this other than just retrying videos a significantly increased amount (which would likely cause other issues for other users) or skipping and unskipping media to retry it manually.

One possible hack might be a "don't attempt to download this item for X days" delay, but that would be manual and per-item so you might as well just skip and unskip if you're going to set that anyway. I'll check the metadata for upcoming live streams to see if there's a sensible date to extract. Last time I checked there wasn't but if there is one available now I can use that to schedule when the download should be attempted.

**Comment by mikeyounge** [comment]
I saw this in the logs so it looks like something is calculated. It might be difficult to assign a time to wait, but 12-24 hours would be reasonable after event begins.

`2024-12-01 22:22:31,036 [tubesync/ERROR] ERROR: [youtube] SClTdM4suco: This live event will begin in 8 hours.`

**Comment by meeb** [comment]
Ah interesting thanks, I'll see if that's available as a sensible date over the internal API rather than just a string error message. If that's available it would be a sensible delay to use.

**Comment by tcely** [comment]
I think I solved this issue for myself before I had gotten around to reading it.

There is now a refresh formats task[^1] that is triggered when things fail to download. Sadly, it can't be much smarter than that.

Also, for upcoming streams[^2] they are marked as skipped until the current time is after the published time claimed for that video. There is a task that runs every hour to mark it as not skipped anymore / update the remaining hours before it is published, as needed.

[^1]: #956
Building on work in: 
#930
[^2]: #676

**Comment by mikeyounge** [comment]
Yes. I think this has been solved. I had the same channel do a similar week of live streams last month and they seemed to all download with no manual intervention. I think we mark this one closed.
